ServerAdministration/PractiaclDocker.md

```diff
@@ -0,0 +1,139 @@
+
+```bash
+sudo usermod -aG docker <USERNAME>@student.aau.dk
+```
+
+If the Docker installation is going to be used in a Docker Swarm at AAU the following changes is need since the AAU network is running on the default Docker subnet.
+
+For local installations use the following.
+
+```bash
+nano /etc/docker/daemon.json
+```
+
+and insert the following:
+
+```json
+{
+  "bip": "10.14.0.1/16",
+  "ipv6": false
+}
+```
+
+For the Swarm overlay network first create the network manualy with the following command:
+
+```bash
+docker network create --subnet 10.10.0.0/16 -o com.docker.network.bridge.enable_icc=false -o com.docker.network.bridge.name=docker_gwbridge 
+```
+
+## Initializing the Swarm
+
+REMEMBER TO CREATE THE NETWORK FIRST
+On the first master server run the following command:
+
+```bash
+docker swarm init
+```
+
+### Adding a manager to the Swarm
+
+REMEMBER TO CREATE THE NETWORK FIRST
+Once the Docker Swarm is initialized used the following command on the first master server to get the join token for the swarm:
+
+```basj
+docker swarm join-token --manager
+```
+
+The output is copied to the new master server.
+
+### Adding a worker to the Swarm
+
+REMEMBER TO CREATE THE NETWORK FIRST
+To add a worker to the Swarm first run the following command on one of the managers:
+
+```bash
+docker swarm join-token worker
+```
+
+The output should be run on the serveres intended to be workers in the Swarm.
+
+Now the Docker Swarm has been created and is ready to serve.
+
+Use the following command to verify the Swarm setup:
+
+```bash
+docker swarm ls
+```
+
+The output should look like this:
+
+| ID                           | HOSTNAME                   | STATUS             | AVAILABILITY       | MANAGER STATUS     | ENGINE VERSION |
+| :--------------------------- | :------------------------- | :----------------- | :----------------- | :----------------- | :------------- |
+| yny9ky6b6zczqrjzxd7sl71k6 *  | giraf-master00.srv.aau.dk  | Ready              | Active             | Leader             | 18.09.3        |
+| 2n08r588w9p8xazc5cm8r6o9o    | giraf-master01.srv.aau.dk  | Ready              | Active             | Reachable          | 18.09.3        |
+| wrr68nqt116tk1rszwvdv1nmk    | giraf-node00.srv.aau.dk    | Ready              | Active             |                    | 18.09.3        |
+| bhh5mitvwzdhzbky1cjne9ffg    | giraf-node01.srv.aau.dk    | Ready              | Active             |                    | 18.09.3        |
+| as7n375y2gwcj5vf4h73h9ron    | giraf-node02.srv.aau.dk    | Ready              | Active             |                    | 18.09.3        |
+| koclcs8nxt0y6qu4ho511la0m    | giraf-node03.srv.aau.dk    | Ready              | Active             |                    | 18.09.3        |
+
+
+## Examples of using a service
+
+Once the Docker Swarm has been set up it can be used to serve the different parts of the project. In the following we will give some examples on how to work with the Swarm.
```

Comment:
```suggestion
Once the Docker Swarm has been set up, it can be used to serve the different parts of the project. In the following, we will give some examples of how to work with the Swarm.
```